projects/batfish/src/main/java/org/batfish/bdp/VirtualRouter.java

```diff
@@ -1088,10 +1183,45 @@ public int propagateBgpRoutes(Map<String, Node> nodes, Map<Ip, Set<String>> ipOw
                         new TreeSet<>(receivedCommunities),
                         new TreeSet<>(receivedClusterList),
                         receivedWeight);
-                _receivedBgpAdvertisements.add(receivedAdvert);
 
-                if (targetRib.mergeRoute(transformedIncomingRoute)) {
-                  numRoutes++;
+                Prefix prefix = remoteRoute.getNetwork();
+                if (oscillatingPrefixes.contains(prefix)) {
+                  if (hasAdvertisementPriority(
+                      remoteRoute,
+                      dependentRoutesIterations,
+                      oscillatingPrefixes,
+                      neighbor,
+                      remoteBgpNeighbor)) {
+                    synchronized (markedPrefixes) {
```

Comment:
was wondering if we should move the synchronization into the markPrefix function.